datasets/race/race.py

```diff
@@ -61,43 +69,36 @@ def _info(self):
 
     def _split_generators(self, dl_manager):
         """Returns SplitGenerators."""
-        # TODO(race): Downloads the data and defines the splits
+        # Downloads the data and defines the splits
         # dl_manager is a datasets.download.DownloadManager that can be used to
-        # download and extract URLs
         dl_dir = dl_manager.download_and_extract(_URL)
+        case = str(self.config.name)
+        if case == "all":
+            case = ""
         return [
             datasets.SplitGenerator(
                 name=datasets.Split.TEST,
                 # These kwargs will be passed to _generate_examples
-                gen_kwargs={
-                    "files": sorted(os.listdir(os.path.join(dl_dir, "RACE/test/high"))),
-                    "filespath": os.path.join(dl_dir, "RACE/test/high"),
-                },
+                gen_kwargs={"train_test_or_eval": os.path.join(dl_dir, f"RACE/test/{case}")},
             ),
             datasets.SplitGenerator(
                 name=datasets.Split.TRAIN,
                 # These kwargs will be passed to _generate_examples
-                gen_kwargs={
-                    "files": sorted(os.listdir(os.path.join(dl_dir, "RACE/train/high"))),
-                    "filespath": os.path.join(dl_dir, "RACE/train/high"),
-                },
+                gen_kwargs={"train_test_or_eval": os.path.join(dl_dir, f"RACE/train/{case}")},
             ),
             datasets.SplitGenerator(
                 name=datasets.Split.VALIDATION,
                 # These kwargs will be passed to _generate_examples
-                gen_kwargs={
-                    "files": sorted(os.listdir(os.path.join(dl_dir, "RACE/dev/high"))),
-                    "filespath": os.path.join(dl_dir, "RACE/dev/high"),
-                },
+                gen_kwargs={"train_test_or_eval": os.path.join(dl_dir, f"RACE/dev/{case}")},
             ),
         ]
 
-    def _generate_examples(self, files, filespath):
+    def _generate_examples(self, train_test_or_eval):
         """Yields examples."""
-        # TODO(race): Yields (key, example) tuples from the dataset
-        for file in files:
-            filepath = os.path.join(filespath, file)
-            with open(filepath, encoding="utf-8") as f:
+        current_path = Path(train_test_or_eval)
+        files_in_dir = [str(f.absolute()) for f in current_path.glob("**/*.txt")]
```

Comment:
```suggestion
        files_in_dir = [str(f.absolute()) for f in sorted(current_path.glob("**/*.txt"))]
```
We use sorted to make the files order deterministic